pkg/prometheus/resource_selector.go

```diff
@@ -830,13 +840,38 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 	return res, nil
 }
 
+// ValidateSelectorRoles validate the roles based on main role.
+func ValidateSelectorRoles(c monitoringv1alpha1.KubernetesSDConfig) error {
+
+	foundSelectorRoles := make(map[monitoringv1alpha1.Role]struct{})
+	allowedSelectors := map[monitoringv1alpha1.Role][]string{
+		RolePod:           {string(RolePod)},
+		RoleService:       {string(RoleService)},
+		RoleEndpointSlice: {string(RolePod), string(RoleService), string(RoleEndpointSlice)},
+		RoleEndpoint:      {string(RolePod), string(RoleService), string(RoleEndpoint)},
+		RoleNode:          {string(RoleNode)},
+		RoleIngress:       {string(RoleIngress)},
+	}
+
+	for _, selector := range c.Selectors {
+
+		if _, ok := foundSelectorRoles[selector.Role]; ok {
+			return fmt.Errorf("duplicated selector role: %s", selector.Role)
+		}
```

Comment:
I have the impression that `foundSelectorRoles` is never populated, therefore line 858 will never return `ok` 🤔

Reply:
If it returns ok then error will come, while if it returns !ok then it should pass, tests are also passing

Reply:
I agree with Arthur, I also tried it as a simple go code here https://go.dev/play/p/Di7AaLX_iW-
currently it accepts the the role pod for role node which it shouldn't.

Take a look at validation here we need to do something similar https://github.com/prometheus/prometheus/blob/92544c00bf9d43eecd7161911d541a3fbe7af8cc/discovery/kubernetes/kubernetes.go#L204-L233